src/conduits/java/com/enderio/conduits/common/items/ConduitProbeItem.java

```diff
@@ -120,19 +137,42 @@ private void handleCopy(ConduitBlockEntity conduitBlock, Direction face, ItemSta
                 typeTag.putBoolean(IS_INSERT, false);
                 typeTag.putBoolean(IS_EXTRACT, false);
             }
-            conduitData.put(Objects.requireNonNull(ConduitType.getKey(conduitType)).toString(), typeTag);
+            conduitData.put(conduitKey, typeTag);
+
+            message.append(Component.literal("\n" + conduitKeyToDisplayName(conduitKey).getString() + ":\n").withStyle(ChatFormatting.UNDERLINE));
+            typeTag.getAllKeys().forEach(key -> {
+                StringBuilder sb = new StringBuilder();
+                sb.append(" - " + key + ": ");
+                if (BOOL_TAG.contains(key)) {
+                     sb.append(typeTag.getBoolean(key));
+                } else if (COLOR_CONTROL_TAG.contains(key)) {
+                    sb.append(ColorControl.values()[typeTag.getInt(key)]);
+                } else if (key.equals(REDSTONE_CONTROL)) {
+                    sb.append(RedstoneControl.values()[typeTag.getInt(key)]);
+                } else {
+                    sb.append(typeTag.get(key));
+                }
+                sb.append("\n");
+                message.append(Component.literal(sb.toString()).withStyle(ChatFormatting.GRAY));
+            });
         });
+        player.sendSystemMessage(TooltipUtil.withArgs(EIOLang.CONDUIT_PROBE_MESSAGE_COPIED, message));
     }
-    
-    public void handlePaste(ConduitBlockEntity conduitBlock, Direction face, ItemStack itemStack) {
+
+    public void handlePaste(ConduitBlockEntity conduitBlock, Direction face, ItemStack itemStack, Player player) {
         CompoundTag tag = itemStack.getTag();
         if (tag == null || !(tag.get(CONDUIT_DATA) instanceof CompoundTag conduitData)) {
             return;
         }
+
+        List<String> pastedConduits = new ArrayList<String>();
         ConduitBundle bundle = conduitBlock.getBundle();
-        bundle.getTypes().forEach(conduitType -> {
-            CompoundTag typeTag = conduitData.getCompound(Objects.requireNonNull(ConduitType.getKey(conduitType)).toString());
-            ConnectionState prevConnectionState = bundle.getConnectionState(face, conduitType);
+        bundle.getTypes().forEach(conduit -> {
+            String conduitKey = ConduitType.getKey(conduit).toString();
+            CompoundTag typeTag = conduitData.getCompound(conduitKey);
+            if (typeTag == null || typeTag.isEmpty()) return;
```

Comment:
And braces here :)

Reply:
There's a few more - checkstyle has caught a lot of them, please scroll through the diff tab to see them all :)

<img width="751" height="180" alt="Image" src="https://github.com/user-attachments/assets/06791a3a-5265-470c-bd49-bdb7023f667c" />